app/models/list.rb

```diff
@@ -1,5 +1,7 @@
 class List < ApplicationRecord
   belongs_to :user
   has_and_belongs_to_many :movies
-  validates_presence_of :name, :description, :public
+  validates_presence_of :name, :description
+  validates :public, inclusion: { in: [true, false] }
+  validates :public, exclusion: { in: [nil] }
```

Comment:
I don't think you need this validation if you have the previous one. Maybe you can check to be sure.